packages/playground/src/components/ssh-key-create-form.tsx

```diff
@@ -0,0 +1,103 @@
+  content: z.string().min(1, { message: 'Please add the public key' })
+})
+
+export type SshKeyFormType = z.infer<typeof formSchema>
+
+export function SshKeyCreateForm() {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors, isValid }
+  } = useForm<SshKeyFormType>({
+    resolver: zodResolver(formSchema),
+    mode: 'onChange',
+    defaultValues: {
+      identifier: '',
+      content: ''
+    }
+  })
+
+  const [isSubmitted, setIsSubmitted] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const handleFormSubmit: SubmitHandler<SshKeyFormType> = data => {
+    setIsSubmitting(true)
+    setTimeout(() => {
+      console.log('SSH key created:', data)
+      reset()
+      setIsSubmitting(false)
+      setIsSubmitted(true)
+      setTimeout(() => setIsSubmitted(false), 2000)
+    }, 2000)
+  }
+
+  const handleCancel = () => {}
+
+  return (
+    <>
+      <form onSubmit={handleSubmit(handleFormSubmit)}>
+        {/* NAME */}
+        <FormFieldSet.Root>
+          <FormFieldSet.ControlGroup>
+            <FormFieldSet.Label htmlFor="identifier" required>
+              New SSH key
+            </FormFieldSet.Label>
+            <Input id="identifier" {...register('identifier')} placeholder="Enter the name" autoFocus />
+            {errors.identifier && (
+              <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+                {errors.identifier.message?.toString()}
+              </FormFieldSet.Message>
+            )}
+          </FormFieldSet.ControlGroup>
+        </FormFieldSet.Root>
+
+        <FormFieldSet.Root>
+          <FormFieldSet.ControlGroup>
+            <FormFieldSet.Label htmlFor="content" required>
+              Public key
+            </FormFieldSet.Label>
+            <Textarea id="content" {...register('content')} />
+            {errors.identifier && (
+              <FormFieldSet.Message theme={FormFieldSet.MessageTheme.ERROR}>
+                {errors.identifier.message?.toString()}
+              </FormFieldSet.Message>
+            )}
+          </FormFieldSet.ControlGroup>
+        </FormFieldSet.Root>
+
+        {/* SUBMIT BUTTONS */}
+        <FormFieldSet.Root>
+          <FormFieldSet.ControlGroup>
+            <ButtonGroup.Root className="justify-end">
+              {!isSubmitted ? (
+                <>
+                  <Button type="button" variant="outline" size="sm" onClick={handleCancel}>
+                    Cancel
+                  </Button>
+                  <Button type="submit" size="sm" disabled={!isValid || isSubmitting}>
+                    {!isSubmitting ? 'Save' : 'Saving...'}
+                  </Button>
+                </>
+              ) : (
+                <Button variant="ghost" type="button" size="sm" theme="success" className="pointer-events-none">
+                  Key Saved&nbsp;&nbsp;
+                  <Icon name="tick" size={14} />
+                </Button>
```

Comment:
nit: Probably would be better to put this text and icon in a `flex` and then put `gap` between them?

Reply:
<img width="202" alt="key" src="https://github.com/user-attachments/assets/c06a928a-46a6-49c5-9281-a5b22a912cf7">


Not sure if it needs a gap?

Should ideally be conformant across all forms-
https://github.com/harness/canary/blob/main/packages/playground/src/pages/sandbox-repo-create-page.tsx#L240